### PR TITLE
feat(mezmo_redact): Support delimited credit card matching

### PR DESCRIPTION
**Pull request by biblicalph**
Update the credit card matching to support delimited card numbers.
Fix bug in the phone number pattern.
Always redact phone number pattern last to avoid conflicts with credit card pattern

### CODE REVIEW

File: src/stdlib/mezmo_redact.rs [hunk 1]
```diff
@@ -113,6 +114,13 @@ impl Function for MezmoRedact {
             })
             .collect::<std::result::Result<Vec<FilterWithRedactor>, _>>()?;
 
+        // The phone number pattern may match parts of credit cards (no delimiters)
+        // To reduce the blast radius, move the phone filter to the last item
+        filters.sort_by(|item1, item2| match (&item1.filter, &item2.filter) {
+            (&Filter::PhoneNumber, _) => Ordering::Equal,
```
Comment (matt-march) [inline src/stdlib/mezmo_redact.rs:120]: does this need to be `Ordering::Greater`?

File: src/stdlib/mezmo_redact.rs [hunk 2]
```diff
@@ -935,7 +944,7 @@ mod test {
             args: func_args![
                 value: value!({
                     "arr": [
-                        "contains 180011111111122, a JCB credit card",
+                        "contains 1800222233334444, a JCB credit card",
```
Comment (matt-march) [inline src/stdlib/mezmo_redact.rs:947]: in the filters list below, might be worth moving the `phone_number` entry to the front of the array, or at least earlier than the `us_social_security_number` one? just to make the `sort_by()` really work hard

File: src/stdlib/mezmo_redact.rs (hunk 1 above)
Comment (biblicalph) [reply src/stdlib/mezmo_redact.rs]: Yes absolutely

File: src/stdlib/mezmo_redact.rs (hunk 2 above)
Comment (biblicalph) [reply src/stdlib/mezmo_redact.rs:967]: Done

**Review by matt-march — APPROVED** [review]
lgtm!

**Pull request merged**